lib/Graph/Graph.cpp

```diff
@@ -1482,6 +1482,58 @@ Node *Function::createWeightedSum(llvm::StringRef name,
   return currAdd;
 }
 
+Node *Function::createBatchBoxCox(llvm::StringRef name, NodeValue data,
+                                  NodeValue lambda1, NodeValue lambda2) {
+  assert((lambda1.dims() == lambda2.dims()) &&
+         "lambda1 and lambda2 must have the same shape");
+  assert((lambda1.getType()->getElementType() == lambda2.getElementType()) &&
+         "lambda1 and lambda2 must have the same element type");
+  assert((lambda1.getType()->getElementType() == data.getElementType()) &&
+         "data and lambdas must have the same element type");
+  assert((lambda1.dims().size() == 1) && "lambda1 and lambda2 must be vectors");
+  assert((data.dims().size() == 2) && "data must be a matrix");
+  assert((data.dims()[1] == lambda1.dims()[0]) &&
+         "data, lambda1 and lambda2 must have the same number of rows");
+
+  // Broadcast lambda1 and lambda2 so that they are both the same size as the
+  // data.
+  auto *BL1 = createBroadcast(name.str() + ".broadcast", lambda1, data.dims(),
+                              /*axis=*/1);
+  auto *BL2 = createBroadcast(name.str() + ".broadcast", lambda2, data.dims(),
+                              /*axis=*/1);
+
+  // Compute data + BL2, which is needed regardless of whether
+  // lambda1 is 0 or not.
+  auto *AN = createAdd(name.str() + ".add", data, BL2);
+
+  // Compute the Box-Cox transform for the lambda1 == 0 case:
+  //    y = ln(x + lambda2)
+
+  // Clip the data to the log below at 1e-6 to avoid taking the logarithm
```

Comment:
This clip value is from Caffe2: https://github.com/pytorch/pytorch/blob/master/caffe2/operators/batch_box_cox_op.cc#L97

Reply:
This could make sense, but I would save it for a different PR, and leave it as is here.